appium-tests/helpers/cameraHelper.js

```diff
@@ -103,11 +109,185 @@ module.exports.getPicture = function (opts, pid) {
     }, opts);
 };
 
-module.exports.checkPicture = function (pid, cb) {
+// verifies taken picture when the promise is resolved,
+// calls a callback with 'OK' if everything is good,
+// calls a callback with 'ERROR: <error message>' if something is wrong
+// note that this function is executed in the context of tested app
+// and not in the context of tests
+module.exports.checkPicture = function (pid, options, cb) {
+    var isIos = cordova.platformId === "ios";
+    var isAndroid = cordova.platformId === "android";
+    // skip image type check if it's unmodified on Android:
+    // https://github.com/apache/cordova-plugin-camera/#android-quirks-1
+    var skipFileTypeCheck = isAndroid &&
+        (!options.quality || options.quality === 100) &&
```

Comment:
@riknoll the truth is quality default is 50 for all platforms as if no quality option is passed, Camera.js set it to 50 (https://github.com/apache/cordova-plugin-camera/blob/master/www/Camera.js#L145). Even if the java code sets the default to 80 it will be 50 (I have tested). Anyway, I'm going to open an issue and set it 50 in the java code too.